Windows cross-compilation fails to build with older mingw32-gcc versions

Cross-compilation on Ubuntu 22/mingw32-gcc 10.3 targeting Windows fails with
```
link_win32.c: In function ‘caml_unix_link’:
link_win32.c:30:21: error: ‘ENOSYS’ undeclared (first use in this function)
   30 |     caml_unix_error(ENOSYS, "link", path2);
      |                     ^~~~~~
link_win32.c:30:21: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [../Makefile.otherlibs.common:171: link_win32.b.o] Error 1
make[2]: *** Waiting for unfinished jobs....
lockf_win32.c: In function ‘caml_unix_lockf’:
lockf_win32.c:73:23: error: ‘EINVAL’ undeclared (first use in this function)
   73 |       caml_unix_error(EINVAL, "lockf", Nothing);
      |                       ^~~~~~

```

I believe this was accidentally introduced by #13447 which removed the include of `errno.h` from these files, but this include is needed for the error constants.

This seems to go away on newer Ubuntu versions (which is why the cross-compilation CI in this repository doesn't catch it), possibly because a standard library header started transitively including errno.h. I temporarily changed the CI to use ubuntu-22.04 instead of ubuntu-latest here, which exposes the issue (and the trivial fix, which I am happy to contribute): https://github.com/WardBrian/ocaml/pull/1

--- a/otherlibs/unix/link_win32.c
+++ b/otherlibs/unix/link_win32.c
@@ -21,6 +21,7 @@
 #include <caml/osdeps.h>
 #include "caml/unixsupport.h"
 #include <windows.h>
+#include <errno.h>
 
 CAMLprim value caml_unix_link(value follow, value path1, value path2)
 {

--- a/otherlibs/unix/lockf_win32.c
+++ b/otherlibs/unix/lockf_win32.c
@@ -20,6 +20,7 @@
 #include <caml/fail.h>
 #include "caml/unixsupport.h"
 #include <stdio.h>
+#include <errno.h>
 #include <caml/signals.h>
 
 #ifndef INVALID_SET_FILE_POINTER

--- a/otherlibs/unix/readlink_win32.c
+++ b/otherlibs/unix/readlink_win32.c
@@ -23,6 +23,7 @@
 #include "caml/unixsupport.h"
 #include <winioctl.h>
 #include <caml/winsupport.h>
+#include <errno.h>
 
 CAMLprim value caml_unix_readlink(value opath)
 {

--- a/otherlibs/unix/system.c
+++ b/otherlibs/unix/system.c
@@ -23,6 +23,7 @@
 #include "caml/unixsupport.h"
 #include <process.h>
 #include <stdio.h>
+#include <errno.h>
 
 CAMLprim value caml_unix_system(value command)
 {

